### ISSUE TITLE
UndeliverableException in BaseObservableOnSubscribe

### ISSUE THREAD

**Issue by mzgreen** [mcharmas/Android-ReactiveLocation#171]
I'm using 2.0 version of the library along with google play services 11.4.2.

I've reproduced this issue only on Huawei P8 Lite. What I'm doing is:
```
 private val locationRequest: LocationRequest?
        get() = LocationRequest.create()
                .setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY)
                .setInterval(5000)

 val location = locationProvider
                    .getUpdatedLocation(locationRequest)
                    .doOnError { Timber.e(it.toString()) }
                    .retry()
                    .observeOn(rxSchedulers.main())
                    .subscribe({},{})
```

Stacktrace:
```
pl.charmas.android.reactivelocation2.observables.BaseObservableOnSubscribe$ApiClientConnectionCallbacks.onConnectionFailed (BaseObservableOnSubscribe.java:110)
com.google.android.gms.common.internal.zzad.zzk (Unknown Source)
com.google.android.gms.common.api.internal.zzbd.zzc (Unknown Source)
com.google.android.gms.common.api.internal.zzar.zze (Unknown Source)
com.google.android.gms.common.api.internal.zzar.onConnectionSuspended (Unknown Source)
com.google.android.gms.common.api.internal.zzbl.onConnectionSuspended (Unknown Source)
com.google.android.gms.common.api.internal.zzw.onConnectionSuspended (Unknown Source)
com.google.android.gms.common.internal.zzab.onConnectionSuspended (Unknown Source)
com.google.android.gms.common.internal.zzh.handleMessage (Unknown Source)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5538)
java.lang.reflect.Method.invoke (Method.java)
java.lang.reflect.Method.invoke (Method.java:372)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:960)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:755)
```

**Comment by Lir10** [comment]
Hi @mzgreen ,

I'm getting also similar errors from ApiClientConnection 
#176 
#175 
Any ideas?

**Comment by Lir10** [comment]
@mzgreen @mcharmas 
Possible solution in #175

**Comment by mcharmas** [comment]
Should be fixed in 2.1. Propagating to maven central. Should be available in few hours.
